storage_controller/src/node.rs

```diff
@@ -116,6 +116,15 @@ impl Node {
         match (self.availability, availability) {
             (Offline, Active(_)) => ToActive,
             (Active(_), Offline) => ToOffline,
+            // Consider the case when the storage controller handles the re-attach of a node
+            // before the heartbeats detect that the node is back online. We still need
+            // [`Service::node_configure`] to migrate attachments back onto the node.
+            // The unsavoury match arm below handles this situation.
+            (Active(lhs), Active(rhs))
+                if lhs == UtilizationScore::worst() && rhs < UtilizationScore::worst() =>
```

Comment:
unsavoury indeed.  Let's find a way to avoid encoding UtilizationScore::worst as a proxy for "activated but hasn't yet heartbeated".

I'm unsure about the "migrate attachments back..." part: `node_configure` currently only does some reconciliations in its `ToActive` branch -- is this comment hinting at something from another PR that would add migration of attachments back onto the node?

Reply:
i wouldn't describe what it does as migrating attachments to this node: it's really just cleaning up after we potentially skipped some reconciliations because the node was offline.  Ordinarily it won't be attaching anything, because when we marked the node offline we'd have changed the intents to not use the offline node (aside from in the case of all nodes going offline)

Reply:
You're right. Migrating attachments back describes an edge case. Reworded in https://github.com/neondatabase/neon/pull/8060/commits/0ec3bc2135de89bbf0bfe25e2f66f36cf808fbfe